start-tests.sh

```diff
@@ -7,106 +7,267 @@
+      log_color "purple" "OPTIONS_HUB_PASSWORD" "is required to login into the ACM console. ${YELLOW}Skipping UI test.${NC}\n"
+      export SKIP_UI_TEST=true
+    fi
+
+    echo -e "Kubeconfig file detected at: $OPTIONS_HUB_KUBECONFIG => copying to ./kube/config/hub-kubeconfig"
+    cp $OPTIONS_HUB_KUBECONFIG ./kube/config/hub-kubeconfig
+    export OPTIONS_HUB_KUBECONFIG=./kube/config/hub-kubeconfig
+
+    # Check to see if there are any kubecontext to be used from the hub cluster kubeconfig.
+    if [[ -z $OPTIONS_HUB_KUBECONTEXT || "$OPTIONS_HUB_KUBECONTEXT" == "null" ]]; then
+      HUB_CLUSTER=($(oc config get-clusters --kubeconfig=$OPTIONS_HUB_KUBECONFIG))
+      export OPTIONS_HUB_KUBECONTEXT=default/${HUB_CLUSTER[1]}/kube:admin
+    fi
+
+    echo -e
+    log_color "cyan" "Switching context to log into Kube API server"
+    oc config use-context --kubeconfig=$OPTIONS_HUB_KUBECONFIG $OPTIONS_HUB_KUBECONTEXT
+    
+    export OPTIONS_HUB_BASEDOMAIN=$(oc whoami --show-server=true | cut -d'.' -f2- | cut -d':' -f1)
+
+    if [[ $OPTIONS_HUB_OC_IDP == "kube:admin" && -z $OPTIONS_HUB_USER ]]; then
+      export OPTIONS_HUB_USER=kubeadmin
+    fi
+
+    log_color "purple" "HUB CLUSTER:" "$OPTIONS_HUB_BASEDOMAIN"
+  fi
+else
+  echo -e "Environment variables detected for hub cluster. Configuring tests to execute with exported variables."
 fi
 
+# Export variables to cypress.
+export CYPRESS_OPTIONS_HUB_BASEDOMAIN=$OPTIONS_HUB_BASEDOMAIN
+export CYPRESS_OPTIONS_HUB_KUBECONTEXT=$OPTIONS_HUB_KUBECONTEXT
+export CYPRESS_OPTIONS_HUB_PASSWORD=$OPTIONS_HUB_PASSWORD
+export CYPRESS_OPTIONS_HUB_USER=$OPTIONS_HUB_USER
+export CYPRESS_OPTIONS_HUB_OC_IDP=$OPTIONS_HUB_OC_IDP
+
+# Export base url for cluster.
+export BASE_URL=https://multicloud-console.apps.$OPTIONS_HUB_BASEDOMAIN
+export CYPRESS_BASE_URL=$BASE_URL
+
+echo -e
+
 log_color "cyan" "Running tests with the following environment:\n"
 log_color "purple" "\tCYPRESS_OPTIONS_HUB_BASEDOMAIN" "\t: $CYPRESS_OPTIONS_HUB_BASEDOMAIN"
 log_color "purple" "\tCYPRESS_OPTIONS_HUB_BASE_URL" "\t: $CYPRESS_BASE_URL"
 log_color "purple" "\tCYPRESS_OPTIONS_HUB_USER" "\t: $CYPRESS_OPTIONS_HUB_USER"
-log_color "purple" "\tCYPRESS_OC_IDP" "\t\t\t: $CYPRESS_OC_IDP\n"
+log_color "purple" "\tCYPRESS_OPTIONS_HUB_OC_IDP" "\t: $CYPRESS_OPTIONS_HUB_OC_IDP\n"
+
+if [[ ! -z $CYPRESS_OPTIONS_HUB_PASSWORD && "$CYPRESS_OPTIONS_HUB_PASSWORD" != "null" ]]; then
+  log_color "cyan" "Logging into Kube API server."
+
+  export KUBECONFIG=./kube/config/hub-kubeconfig
+  touch $KUBECONFIG
+
+  oc login --server=https://api.${CYPRESS_OPTIONS_HUB_BASEDOMAIN}:6443 -u $CYPRESS_OPTIONS_HUB_USER -p $CYPRESS_OPTIONS_HUB_PASSWORD --insecure-skip-tls-verify
+  export OPTIONS_HUB_KUBECONFIG=$KUBECONFIG
+
+  if [[ -f $OPTIONS_HUB_KUBECONFIG ]]; then
+    echo -e "Succesfully detected hub cluster kubeconfig.\n"
+  else
+    echo -e "Failed to create or locate hub cluster kubeconfig.\n"
+  fi
```

Comment:
If the `hub` cluster password is available, we can log into the `hub` with the `oc` command. Before we log in, we need to set the `kubeconfig` file to our set path for testing. We can check to see if the file was created and found successful.